Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -433,6 +440,9 @@ public struct PackageBuilder {
                     if dependency.type != .library {
                         throw ModuleError.executableAsDependency(module: module.name, dependency: name)
                     }
+                    guard module.name != name else {
+                        throw ModuleError.cycleDetected("the target \(name) can't have a dependency on itself.")
+                    }
```

Comment:
This isn't actually checking for cycles, though?

My guess is this Package.swift blows up to:

targets: [Target(name: "A", dependencies: ["B"]), Target(name: "B", dependencies: "A"])